### PR TITLE
Allow program managers to access workshops assigned to their regional partner,

### PR DESCRIPTION
**Pull request by aoby**
… but not created by them

### CODE REVIEW

**Comment by aoby** [comment]
[Regional Partners see their own workshops and workshops created by a Facilitator for them](https://trello.com/c/VuBfsm2K/442-regional-partners-see-their-own-workshops-and-workshops-created-by-a-facilitator-for-them)

File: dashboard/app/models/ability.rb [hunk 1]
```diff
@@ -107,9 +107,7 @@ def initialize(user)
         end
         can [:read, :start, :end, :workshop_survey_report, :summary, :filter], Pd::Workshop, facilitators: {id: user.id}
         can [:read, :update], Pd::Workshop, organizer_id: user.id
-        can :create, Pd::Workshop do |_|
-          Pd::CourseFacilitator.exists?(facilitator: user, course: Pd::Workshop::COURSE_CSF)
-        end
+        can :create, Pd::Workshop if Pd::CourseFacilitator.exists?(facilitator: user, course: Pd::Workshop::COURSE_CSF)
```
Comment (aoby) [inline dashboard/app/models/ability.rb:110]: @mehalshah, this is redundant with your work in https://github.com/code-dot-org/code-dot-org/pull/20963, and can be replaced by that. I added this when I discovered the issue yesterday, and added a [test](https://github.com/code-dot-org/code-dot-org/pull/20973/files#diff-6b83de5bff687d68cbe6468827b18b01R366)

File: dashboard/app/models/pd/workshop.rb [hunk 2]
```diff
@@ -197,7 +197,7 @@ class Pd::Workshop < ActiveRecord::Base
   before_save :process_location, if: -> {location_address_changed?}
   auto_strip_attributes :location_name, :location_address
 
-  before_save :assign_regional_partner, if: -> {organizer_id_changed?}
+  before_save :assign_regional_partner, if: -> {organizer_id_changed? && !regional_partner_id?}
```
Comment (mehalshah) [inline dashboard/app/models/pd/workshop.rb:200]: Also redundant with that change - but lets see which gets merged first

Comment (aoby) [reply dashboard/app/models/pd/workshop.rb:200]: The race is on...

**Pull request merged**